explore/src/main/scala/explore/TopBar.scala

```diff
@@ -35,46 +36,50 @@ import react.semanticui.modules.dropdown.DropdownItem
 import react.semanticui.modules.dropdown.DropdownMenu
 import react.semanticui.shorthand._
 import react.semanticui.views.item.Item
+import typings.loglevel.mod.LogLevelDesc
 
 final case class TopBar(
-  user:     User,
-  onLogout: IO[Unit]
-) extends ReactProps[TopBar](TopBar.component)
+  user:        User,
+  preferences: ExploreLocalPreferences,
+  onLogout:    IO[Unit]
+) extends ReactFnProps[TopBar](TopBar.component)
 
 object TopBar {
   type Props = TopBar
 
-  protected case class State(copied: Boolean = false, theme: Theme) {
-    def flip: State =
-      if (theme === Theme.Dark) copy(theme = Theme.Light) else copy(theme = Theme.Dark)
-  }
-
-  object State {
-    val copied = Focus[State](_.copied)
-  }
-
   implicit val propsReuse: Reusability[Props] = Reusability.by(_.user)
-  implicit val stateReuse: Reusability[State] = Reusability.derive
 
   def currentTheme: Theme =
     if (dom.document.body.classList.contains(Theme.Light.clazz.htmlClass))
       Theme.Light
     else
       Theme.Dark
 
-  private val component =
-    ScalaComponent
-      .builder[TopBar]
-      .initialState(State(false, currentTheme))
-      .render { $ =>
-        val p            = $.props
-        val currentTheme = $.state.theme
+  def flipTheme(theme: Theme): Theme =
+    if (theme === Theme.Dark) Theme.Light else Theme.Dark
 
+  private val component =
+    ScalaFnComponent
+      .withHooks[Props]
+      // copied
+      .useState(false)
+      // theme
+      .useState(currentTheme)
+      .renderWithReuse { (props, copied, theme) =>
         AppCtx.using { implicit appCtx =>
-          val role = p.user.role
+          val role = props.user.role
 
           def logout: IO[Unit] =
-            appCtx.sso.logout >> p.onLogout
+            appCtx.sso.logout >> props.onLogout
+
+          val level = props.preferences.level
+
+          def setLogLevel(l: LogLevelDesc): Callback =
+            ExploreLocalPreferences
+              .storePreferences[IO](
+                props.preferences.copy(level = l)
+              )
+              .runAsync *> Callback(window.location.reload(false))
```

Comment:
I would suggest doing `(....storePreferences[IO](...) *> IO(window.location.reload(false))).runAsync`.

That way we make sure `storePreferences` completes before reloading.